types/auth0/index.d.ts

```diff
@@ -1121,11 +1126,11 @@ export class AuthenticationClient {
     requestSMSCode(data: RequestSMSOptions): Promise<any>;
     requestSMSCode(data: RequestSMSOptions, cb: (err: Error, message: string) => void): void;
 
-    verifyEmailCode(data: VerifyOptions): Promise<any>;
-    verifyEmailCode(data: VerifyOptions, cb: (err: Error, message: string) => void): void;
+    verifyEmailCode(data: VerifyEmailOptions): Promise<any>;
+    verifyEmailCode(data: VerifyEmailOptions, cb: (err: Error, message: string) => void): void;
 
-    verifySMSCode(data: VerifyOptions): Promise<any>;
-    verifySMSCode(data: VerifyOptions, cb: (err: Error, message: string) => void): void;
+    verifySMSCode(data: VerifySMSOptions): Promise<any>;
+    verifySMSCode(data: VerifySMSOptions, cb: (err: Error, message: string) => void): void;
```

Comment:
the docs says data should support `otp` in some scenarios:
https://github.com/auth0/node-auth0/blob/3607d33cb83651e5d3405a2a449a162724bdfb3a/src/auth/index.js#L272
https://github.com/auth0/node-auth0/blob/3607d33cb83651e5d3405a2a449a162724bdfb3a/src/auth/index.js#L308

Reply:
Ah, you are correct.  Sending `password` is accepted, but deprecated further down the chain. So using it here will produce a warning.
https://github.com/auth0/node-auth0/blob/3607d33cb83651e5d3405a2a449a162724bdfb3a/src/auth/PasswordlessAuthenticator.js#L108

Reply:
@peterblazejewicz do you think we should have both and have type completion be suggesting deprecated inputs?

Reply:
yes, please. Both are needed until removed

Reply:
I'm not sure how DefinitelyTyped prefers handling these things.  It seems like it really should be the library maintaining what shows up as deprecated or not, but you could make that argument for the entire DefinitelyTyped
 the project, it would be great if the library maintainers wrote their own types (even better just adopted TS).

Anyway, I arrived at this
   
 ```verifySMSCode(data: VerifySMSOptions | VerifySMSOptionsDeprecated): Promise<any>;```

Which creates a separate type for each instance instead of making both `password` and `otp` optional.